### DISCUSSION TITLE
Using mitiq ZNE with `qiskit_runtime.Sampler`.

### DISCUSSION THREAD

**Discussion by Z035** [unitaryfund/mitiq#2035] [Q&A] [answered]
We are running a `SamplingVQE` with qiskits new `qiskit_ibm_runtime.Sampler `. The `SampleVQE` creates a jobs for every iteration with a unique ansatz. In each job, the `sampler` run and measure the circuit multiple times, where the result is a distribution of class `QuasiDistribution`. (see code below)
<img width="758" alt="截屏2023-09-29 下午2 18 04" src="https://github.com/unitaryfund/mitiq/assets/123568502/ed4a5308-9648-4637-835a-96491f5e1037">

My question is this: I want to incorperate mitiq ZNE with the `Sampler`. But since ZNE returns an expectation value instead of a distribution, how should I do it?

**Comment by andreamari** [comment, marked as answer]
Unless you apply strong customization of the source code, `mitiq.zne` acts on expectation values and is not able to act on intermediate  quasi-distributions.

In practice, in order to use `mitiq.zne` you need an executor, i.e., a function mapping circuits (without symbolic parameters) to expectation values. See e.g. this [Qiskit example](https://mitiq.readthedocs.io/en/stable/examples/ibmq-backends.html).

**Reply by natestemen** [reply]
We have talked about adding the functionality to mitigate distributions before. I guess we could run the noise scaled circuits as normal, but instead of receiving expectation values to extrapolate, we get counts which we use to extrapolate by converting the counts to probabilities. @andreamari do you think this would be useful?

**Reply by andreamari** [reply]
It could be useful. But it would also require a quite radical change in the structure of Mitiq code, since now most techniques are based on post-processing of expectation values.
